### PR TITLE
Feat/v2 workspaces

### PR DESCRIPTION
**Pull request by lfbrehm**
# Workspaces
- Adds missing gRPC requests
# Hooks
- Adds template support
# General
- Fixes some sync issues
- Adds partial unique constraints on db-level for internal relations

### CODE REVIEW

File: src/database/dsls/object_dsl.rs [hunk 1]
```diff
@@ -444,7 +444,18 @@ impl Object {
...
         Ok(())
     }
-
+    pub async fn batch_claim(
+        user_id: &DieselUlid,
+        objects: &Vec<DieselUlid>,
+        client: &Client,
+    ) -> Result<()> {
+        let query = "UPDATE objects 
+            SET data_class = ('CONFIDENTIAL'), created_by = $1
```
Comment (das-Abroxas) [inline src/database/dsls/object_dsl.rs:453]: But the claimed objects do not meet the requirements for our definition of the data class 'Confidential'?

File: src/grpc/hooks.rs [hunk 2]
```diff
@@ -184,10 +186,45 @@ impl HooksService for HookServiceImpl {
...
+        )
+        .ok_or_else(|| tonic::Status::not_found("Hook not found"))?;
+
+        if user != hook.owner {
+            return Err(tonic::Status::unauthenticated("Access not allowed"));
+        }
+        tonic_internal!(
+            self.database_handler.append_project_to_hook(request).await,
+            "HookCallback failed"
+        );
```
Comment (das-Abroxas) [inline src/grpc/hooks.rs:227]: This could be moved into the middlelayer.

**Review by das-Abroxas — APPROVED** [review]
Two comments that can be addressed before merge.

File: src/database/dsls/object_dsl.rs (hunk 1 above)
Comment (lfbrehm) [reply src/database/dsls/object_dsl.rs:453]: You're right, this should be set to `PRIVATE`.

**Review by St4NNi — APPROVED** [review]
gogo!

**Pull request merged**